helmfile.d/charts/prometheus-alerts/templates/alerts/opensearch.yaml

```diff
@@ -67,13 +67,13 @@ spec:
         summary: Opensearch Cluster Red (instance {{`{{ $labels.instance }}`}})
         description: Opensearch Cluster is in a Red status VALUE = {{`{{ $value }}`}}  LABELS = {{`{{ $labels }}`}}
     {{- range $prefixes := .Values.osIndexAlerts }}
-    - alert: {{ $prefixes.prefix | title }}SizeIncreasedOverLimit
-      expr: elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system",index=~"{{ $prefixes.prefix }}.+"} / (1024^2) > {{ $prefixes.alertSizeMB }}
+    - alert: OpenSearch{{ $prefixes.prefix | trimSuffix "-" | title }}IndexSizeOverLimit
+      expr: (elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"} / (1024 ^ 2) > {{ $prefixes.alertSizeMB }}) * clamp(rate(elasticsearch_indices_docs_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"}[1h]) > 0, 1, 1)
```

Comment:
I'm not very proficcient in PromQL so could there be some explanation of the expression

The first part I get 
`(elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"} / (1024 ^ 2)`
but it would still evaluate against all indices part of the regex and not just today's index right?

The second part is a bit more unclear to me

`{{ $prefixes.alertSizeMB }}) * clamp(rate(elasticsearch_indices_docs_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"}[1h]) > 0, 1, 1)`
```

maybe it is possible to mention how the magical `clamp` function works as to only have it evaluate the latest index?

Reply:
So basically, the first part `(elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"} / (1024 ^ 2) > {{ $prefixes.alertSizeMB }})` checks whether the indices size are larger than the predefined `alertSize` and it returns the value of it. 

Then the second part `clamp(rate(elasticsearch_indices_docs_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"}[1h]) > 0, 1, 1)` checks if there is any changes in the document count. If there is the `clamp` function will then clamps the sample values of all elements in the vector to have a lower limit of min and an upper limit of max which in this case it convert those values to 1. 

So basically, the whole expression returns the Active primary shard size for the given index if it is more than the `alertSizeMB`.

Reply:
Thanks for the explanation. I will have to read it a couple of times :sweat_smile: